src/unitxt/operator.py

```diff
@@ -351,22 +353,42 @@ def process(self, instance: dict, multi_stream: MultiStream) -> dict:
         pass
 
 
-class SequntialOperator(MultiStreamOperator):
+class SequentialOperator(MultiStreamOperator):
     """
     A class representing a sequential operator in the streaming system.
 
     A sequential operator is a type of `MultiStreamOperator` that applies a sequence of other operators to a `MultiStream`. It maintains a list of `StreamingOperator`s and applies them in order to the `MultiStream`.
     """
 
+    max_steps = None
+
     steps: List[StreamingOperator] = field(default_factory=list)
 
+    def num_steps(self) -> int:
+        return len(self.steps)
+
+    def set_max_steps(self, max_steps):
+        assert (
+            max_steps <= self.num_steps()
+        ), f"Max steps requested ({max_steps}) is larger than defined steps {self.num_steps()}"
+        assert max_steps >= 1, f"Max steps requested ({max_steps}) is less than 1"
+        self.max_steps = max_steps
+
+    def get_last_step_description(self):
+        last_step = self.max_steps - 1 if not self.max_steps is None else len(self.steps) - 1
+        description = str(self.steps[last_step])
+        return re.sub(r"\w+=None, ", "", description)
+
+    def _get_max_steps(self):
+        return self.max_steps if not self.max_steps is None else len(self.steps)
```

Comment:
```suggestion
        return self.max_steps if self.max_steps is not None else len(self.steps)
```

Reply:
i think also just `self.max_steps if self.max_steps else len(self.steps)` should work